classes/DataWarehouse/Access/Usage.php

```diff
@@ -1107,4 +1115,68 @@ private function generateFilename($title, $start_date, $end_date, $isTimeseries)
             . ($isTimeseries ? 'timeseries' : 'aggregate')
         ;
     }
+
+    /**
+     * Attempts to translate the $usageFilterValue which is either a quoted string ( single or double ),
+     * numeric string, or number via a sql query, based on $usageFilterType. If a non-quoted string
+     * is supplied it will be treated as a number. This affects the logic of what values are returned.
+     * Currently supported $usageFilterType values and the tables ( columns ) they lookup values
+     * in are:
+     *
+     *   - 'pi':       modw.systemaccount ( username )
+     *   - 'resource': modw.resourcefact  ( code )
+     *   - 'project':  modw_cloud.account ( display )
+     *
+     * If the $usageFilterValue is numeric or the $usageFilterType is not one of those currently
+     * supported then array($usageFilterValue) is returned.
+     *
+     * @param string $usageFilterType the 'type' of filter to translate. Currently supported
+     *                                 values: pi, resource, project
+     * @param string|int  $usageFilterValue the value to be translated
+     * @return array of the translated values.
+     * @throws Exception if there is a problem connecting to the db or executing a query.
+     */
+    private function translateFilterValue($usageFilterType, $usageFilterValue)
+    {
+
+        $query = null;
+
+        switch ($usageFilterType) {
+            case 'pi':
+                // This query may return multiple values
+                $query = "SELECT DISTINCT sa.person_id AS value FROM modw.systemaccount sa WHERE sa.username = :value;";
+                break;
+            case 'resource':
+                $query = "SELECT id AS value FROM modw.resourcefact WHERE code = :value";
+                break;
+            case 'project':
+                $query = "SELECT account_id AS value FROM modw_cloud.account WHERE display = :value";
+                break;
+        }
+
+        $filterValueIsString = preg_match('/^[\'"].*[\'"]$/', $usageFilterValue) === 1;
+
+        // We only attempt translation if we support the `$usageFilterType` provided and
+        // the `$usageFilterValue` is a quoted string.
+        if ($query !== null && $filterValueIsString) {
+            $value = trim($usageFilterValue, '\'"');
+
+            $db = DB::factory('database');
+
+            $stmt = $db->prepare($query);
+            $stmt->execute(array(':value' => $value));
+
+            $rows = $stmt->fetchAll(PDO::FETCH_COLUMN, 0);
+
+            // We need to test if there was an error ( bool returned ) because count(bool) === 1.
+            if ($rows !== false && count($rows) > 0) {
+                return $rows;
+            }
+
+            // If a string was provided but no id(s) were found then exception out.
+            throw new Exception(sprintf("Invalid filter value detected: %s", $usageFilterValue));
```

Comment:
One last (nitpicky) thing that I should have mentioned before - lets add the type in there as well:
```suggestion
            throw new Exception(sprintf("Invalid filter value detected for '%s': %s", $usageFilterType, $usageFilterValue));
```